sdk/go/pulumi-language-go/main.go

```diff
@@ -328,6 +347,22 @@ func (m *modInfo) getPlugin() (*pulumirpc.PluginDependency, error) {
 	return plugin, nil
 }
 
+// Reads and parses the go.mod file from the current working directory.
+func (host *goLanguageHost) loadGomod() (*modfile.File, error) {
+	path := filepath.Join(host.cwd, "go.mod")
```

Comment:
Shouldn't this be using the program's pwd, not the plugins cwd?

Reply:
Nah it can be separate, happy with the idea of derisking this by maintaining current behaviour even if it does seem wrong